crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -239,34 +237,36 @@ attributes.
 
 ```py
 class C:
-    variable_with_class_default: str = "value in class body"
+    variable_with_class_default1: str = "value in class body"
+    variable_with_class_default2 = 1
 
     def instance_method(self):
-        self.variable_with_class_default = "value set in instance method"
+        self.variable_with_class_default1 = "value set in instance method"
 
-reveal_type(C.variable_with_class_default)  # revealed: str
+reveal_type(C.variable_with_class_default1)  # revealed: str
+reveal_type(C.variable_with_class_default2)  # revealed: Literal[1]
 
 c_instance = C()
 
-# TODO: should be `str`
-reveal_type(c_instance.variable_with_class_default)  # revealed: @Todo(instance attributes)
+reveal_type(c_instance.variable_with_class_default1)  # revealed: str
+reveal_type(c_instance.variable_with_class_default2)  # revealed: Unknown | Literal[1]
```

Comment:
Is it correct that we want `Unknown | Literal[1]` for the instance attribute access, but `Literal[1] for the class variable access? (above)

Reply:
No, I think we should probably add the union with `Unknown` for class attributes also; the same principles apply. (Though perhaps less strongly in practice, since mutation of class attributes is much less common -- but pyright and mypy both allow it, and we should too since it's allowed at runtime.)

Reply:
Ok, I added an additional test with a TODO for this case.